presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -299,7 +303,55 @@ public ColumnMetadata getColumnMetadata(ConnectorTableHandle tableHandle, Connec
     @Override
     public ConnectorTableHandle createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
     {
-        throw new UnsupportedOperationException();
+        if (!allowCreateTable) {
+            throw new PrestoException(PERMISSION_DENIED, "Creating tables is disabled in this Hive catalog");
+        }
+
+        checkArgument(!isNullOrEmpty(tableMetadata.getOwner()), "Table owner is null or empty");
+
+        SchemaTableName schemaTableName = tableMetadata.getTable();
+        String schemaName = schemaTableName.getSchemaName();
+        String tableName = schemaTableName.getTableName();
+
+        ImmutableList.Builder<String> columnNamesBuilder = ImmutableList.builder();
+        ImmutableList.Builder<Type> columnTypesBuilder = ImmutableList.builder();
+
+        getColumnInfo(tableMetadata, columnNamesBuilder, columnTypesBuilder);
+
+        List<String> columnNames = columnNamesBuilder.build();
+        List<Type> columnTypes = columnTypesBuilder.build();
+
+        ImmutableList.Builder<FieldSchema> partitionKeysBuilder = ImmutableList.builder();
+        ImmutableList.Builder<FieldSchema> columnsBuilder = ImmutableList.builder();
+
+        buildColumns(columnNames, columnTypes, columnsBuilder, partitionKeysBuilder, tableMetadata);
+
+        Path targetPath = getTargetPath(schemaName, tableName, schemaTableName);
+
+        SerDeInfo serdeInfo = new SerDeInfo();
+        serdeInfo.setName(tableName);
+        serdeInfo.setSerializationLib(this.hiveStorageFormat.getSerDe());
```

Comment:
HiveStorageFormat hiveStorageFormat = getHiveStorageFormat(session, this.hiveStorageFormat);